pkg/minishift/registration/redhat.go

```diff
@@ -51,17 +52,36 @@ func (registrator *RedHatRegistrator) CompatibleWithDistribution(osReleaseInfo *
 }
 
 func (registrator *RedHatRegistrator) Register(param *RegistrationParameters) error {
-	if output, err := registrator.SSHCommand("sudo -E subscription-manager version"); err != nil {
+	output, err := registrator.SSHCommand("sudo -E subscription-manager version")
+	if err != nil {
 		return err
-	} else {
-		if strings.Contains(output, "not registered") {
+	}
+	if strings.Contains(output, "not registered") {
+		for i := 1; i < 4; i++ {
+			if param.Username == "" {
+				param.Username = param.GetUsernameInteractive("Username to register to RHSM")
```

Comment:
I am fine with RHSM, but I think we should spell it out. Acronyms are always a pitfall for users. What's about:

"Username for Red Hat Subscription Management (RHSM):"

As said before, this still does not solve the ambiguity for RedHat Developers. Maybe:

"Red Hat Developers or Red Hat Subscription Management (RHSM) username:"

This somehow covers these two bases.